CenterForOpenScience/osf.io
# Ensures DraftNode permission checks are routed to DraftRegistration

## Purpose
The FileDetail view (`/files/5e31b37b6f2a41000caebc82/`) for a file on a DraftNode was failing for users that did not initiate the DraftRegistration but had permission to the DraftRegistration. This is because permissions for the file were being checked on the DraftNode (where only the DraftRegistration initiator had permissions) rather than the DraftRegistration, where contributors are added. 

## Changes
* Added a `if` branch to the `ContributorOrPublic` permissions class to change the object to check permissions on if the original object is a DraftNode (which it is for the `target` of a file)
* Removes `api.draft_node.permissions.ContributorOnDraftRegistration` permissions class
  * This class is now redundant. While having a more specific permissions class such as ContributorOnDraftRegistration is ideal, it cannot be added as a permissions class to the FileDetail endpoint to address this issue because `ContributorOrPublic` is still a necessary permissions class for files on other types of objects and permissions will fail on `ContributorOrPublic`. So, the modification in this PR to `ContributorOrPublic` is necessary and effectively replaces `ContributorOnDraftRegistration`.

## QA Notes

Please make verification statements inspired by your code and what your code touches.
- Verify that contributors on no-project draft registrations are able to view files
- Verify that other files permissions have not been aversely affected by these changes (existing tests should help verify this)

What are the areas of risk?
File Permissions

Any concerns/considerations/questions that development raised?

## Documentation
n/a

## Side Effects
n/a

## Ticket
Pre-release fix
